plugins/node/opentelemetry-instrumentation-restify/src/instrumentation.ts

```diff
@@ -0,0 +1,166 @@
+          if (isWrapped(Server.prototype[name])) {
+            this._unwrap(Server.prototype, name);
+          }
+          this._wrap(Server.prototype, name as keyof Server, this._middlewarePatcher.bind(this));
+        }
+        return moduleExports;
+      },
+      (moduleExports, moduleVersion) => {
+        diag.debug(`Removing patch for ${MODULE_NAME}@${moduleVersion}`);
+        this._isDisabled = true;
+        if (moduleExports) {
+          const Server: any = moduleExports;
+          for (const name of RESTIFY_METHODS) {
+            this._unwrap(Server.prototype, name as keyof Server);
+          }
+          for (const name of RESTIFY_MW_METHODS) {
+            this._unwrap(Server.prototype, name as keyof Server);
+          }
+        }
+      }
+    ));
+
+    return module;
+  }
+
+  private _middlewarePatcher (original: Function, methodName?: string) {
+    const instrumentation = this;
+    return function (this: Server, ...handler: restify.RequestHandler[][] | restify.RequestHandler[]) {
+      return original.call(this, instrumentation._handlerPatcher({ type: types.LayerType.MIDDLEWARE, methodName }, handler));
+    };
+  }
+
+  private _methodPatcher (original: Function, methodName?: string) {
+    const instrumentation = this;
+    return function (this: Server, path: any, ...handler: restify.RequestHandler[][] | restify.RequestHandler[]) {
+      return original.call(this, path, ...instrumentation._handlerPatcher({ type: types.LayerType.REQUEST_HANDLER, path, methodName }, handler));
+    };
+  }
+
+  // will return the same type as `handler`, but all functions recusively patched
+  private _handlerPatcher (metadata: types.Metadata, handler: restify.RequestHandler | restify.RequestHandler[] | restify.RequestHandler[][]): any {
+    if (Array.isArray(handler)) {
+      return (handler as restify.RequestHandler[]).map((handler: any) => this._handlerPatcher(metadata, handler));
+    }
+    if (typeof handler === 'function') {
+      return (req: types.Request, res: restify.Response, next: restify.Next) => {
+        if (this._isDisabled) {
+          return handler(req, res, next);
+        }
+        // const parentSpan = api.getSpan(api?.context?.active());
+        const route = (typeof req.getRoute === 'function' ? req.getRoute()?.path : req.route?.path);
+        const spanName = route || metadata.methodName || metadata.type;
+        const attributes = {
+          [types.CustomAttributeNames.VERSION]: this._moduleVersion || 'n/a',
+          [types.CustomAttributeNames.TYPE]: metadata.type,
+          [types.CustomAttributeNames.METHOD]: metadata.methodName,
+          [HttpAttribute.HTTP_ROUTE]: route,
+        }
+        const span = this.tracer.startSpan(spanName, {
+          attributes,
+        }, api.context.active()); // TODO: <- with this I intend to find and attach all consecutive handlers to HTTP span
+        // .. but instead all spans are attached to the previous handler's span.
```

Comment:
I actually don't specifically mind the fact that the parent spans are automatically set to the (already ended) span of the last handler, but there should be a way, for at least the instrumentation of the first handler for the every request to access the HTTP span it's tied to, right? How?